frameworks/fmi2api/src/main/java/org/intocps/maestro/framework/fmi2/api/mabl/scoping/ScopeFmi2Api.java

```diff
@@ -219,6 +216,92 @@ protected StringVariableFmi2Api store(Supplier<String> nameProvider, String valu
                 newAIdentifierExp(name));
     }
 
+    /**
+     *
+     * @param identifyingName the name of the array
+     * @param mdArray non-jagged multidimensional array.
+     * @param <V> data type
+     * @return an ArrayVariable representing the multidimensional array
+     */
+    private <V> ArrayVariableFmi2Api<V> createMDArray(String identifyingName, V[] mdArray) {
```

Comment:
what is MD. If multi dimensional then why would we have two different methods. It would probably be easier to identify if we have one for creating arrays that has overloads according to the array you want to create.

Reply:
MD is indeed multi dimensional. I couldn't find any method to overload for creating multi dimensional arrays but I could very well have missed it. As I see it the array is created as part of the generic 'store' method but it did not handle multi dimensional arrays.